knack/cli.py

```diff
@@ -101,6 +101,8 @@ def __init__(self,
 
         self.only_show_errors = self.config.getboolean('core', 'only_show_errors', fallback=False)
         self.enable_color = self._should_enable_color()
+        # Init colorama only in Windows legacy terminal
+        self._should_init_colorama = self.enable_color and os.name == 'nt' and not is_modern_terminal()
```

Comment:
Why not `(self.enable_color or _KNACK_TEST_FORCE_ENABLE_COLOR) and ...`? Does that mean knack test always use colorama no matter it's modern terminal or legacy terminal?

Reply:
> Why not `(self.enable_color or _KNACK_TEST_FORCE_ENABLE_COLOR) and ...`? 

The action scope of mocked `_KNACK_TEST_FORCE_ENABLE_COLOR` is `CLI.invoke`. If `_KNACK_TEST_FORCE_ENABLE_COLOR` is used in `CLI.__invoke__`, `CLI.invoke` won't be affected and won't have colorama enabled.

https://github.com/microsoft/knack/blob/cd590312b9b4ea1402d7a0a76814f6ca4da7fab8/tests/util.py#L36-L37

> Does that mean knack test always use colorama no matter it's modern terminal or legacy terminal?

Yes.

Reply:
Dropped `_KNACK_TEST_FORCE_ENABLE_COLOR`. Now strip control sequences by knack itself:

https://github.com/microsoft/knack/blob/e5e0529db43a2dbef50096237d82d3003506a34f/tests/util.py#L57-L58

Reply:
huge work😂